src/http/routes.cc

```diff
@@ -116,8 +118,10 @@ future<std::unique_ptr<http::reply> > routes::handle(const sstring& path, std::u
             return r.handle_exception(_general_handler);
         } catch (const redirect_exception& _e) {
             rep.reset(new http::reply());
-            rep->add_header("Location", _e.url).set_status(_e.status()).done(
-                    "json");
+	    for (const auto& [header, value] : _e.headers()) {
+	      rep->add_header(header, value);
+	    }
+	    rep->set_status(_e.status()).done("json");
```

Comment:
I feel it's a bit too opaque in the usage site.
Before this change, I could see that code set `Location` with the content of the exception.

maybe, we could have `redirect_exception::to_reply() const->http::reply` and encapsulate the logic there
`*rep=e.to_reply();`

Reply:
I like the `to_reply()` change! 

I was originally going to add `retry_after` as another field to `redirect_exception`, but then I thought that eventually there may be other use cases or headers we would want to add to exceptions derived from `httpd::base_exception` in the future, which is why I added the headers map to `base_exception`.

Do you also prefer the explicit variables in the derived classes over the map in base?

Thanks for the comment!

Reply:
looking only at this use, I would say that we could live with just adding an optional<int> member field. 
did you have other use in mind for the first commit? 
I wonder in that case if we could just embed an http::reply (but I don't have the repo open right now, might be a silly suggestion)

Reply:
No, I don't have another specific use case in mind at this time. 

I'll revert the first commit and add the optional field to `redirect_exception` instead with a `to_reply()` method added.